File: src/main/java/io/github/toberocat/guiengine/utils/BStatsCollector.java
package io.github.toberocat.guiengine.utils;

import io.github.toberocat.guiengine.GuiEngineApiPlugin;
import org.bstats.bukkit.Metrics;
import org.jetbrains.annotations.NotNull;

/**
 * Created: 21.07.2023
 *
 * @author Tobias Madlberger (Tobias)
 */
public class BStatsCollector {
    public BStatsCollector(@NotNull GuiEngineApiPlugin plugin) {
        Metrics metrics = new Metrics(plugin, 19167);
    }
}

File: src/main/kotlin/io/github/toberocat/guiengine/utils/BStatsCollector.kt
package io.github.toberocat.guiengine.utils

import io.github.toberocat.guiengine.GuiEngineApiPlugin
import org.bstats.bukkit.Metrics

/**
 * BStatsCollector is a utility class responsible for collecting usage statistics using bStats.
 *
 *
 * Created: 21.07.2023
 * Author: Tobias Madlberger (Tobias)
 */
class BStatsCollector(plugin: GuiEngineApiPlugin) {
    /**
     * Creates a new BStatsCollector and initializes the bStats metrics collection for the plugin.
     *
     * @param plugin The GuiEngineApiPlugin instance to collect metrics for.
     */
    init {
        Metrics(plugin, 19167)
    }
}
